src/impls/bool.rs

```diff
@@ -1,11 +1,12 @@
 use crate::prelude::*;
 
 impl StableHash for bool {
-    fn stable_hash<H: StableHasher>(&self, sequence_number: H::Seq, state: &mut H) {
+    #[inline]
+    fn stable_hash<H: StableHasher>(&self, field_address: H::Addr, state: &mut H) {
         profile_method!(stable_hash);
 
         if *self {
-            state.write(sequence_number, &[]);
+            state.write(field_address, &[]);
```

Comment:
This has consequences for how you can change struct definitions and still get the same hash; e.g., you can add a boolean field and as long as its value is `false`, you get the same hash as if the struct did not have the boolean field. Similar for empty strings and `None` options, but not for ints, where adding a field with value `0` still changes the hash.

It should be clearly documented (probably at the crate level or for `StableHasher`/`StableHash`) what changes to structs do not change the hash, and which ones do.

Reply:
For ints adding a value with `0` does not change the hash. No default values change the hash by design, including `vec![]`, etc. This is to support backward compatibility so we can add features to graph-node without changing PoIs. One weird case is that `Option(0)` _does_ change the hash but that's not possible to fix without creating a collision with `None`.

Why do you think adding a field with `0` changes the hash?

I will add some documentation to the crate, good suggestion.